src/esp32cam.h

```diff
@@ -29,33 +42,194 @@ class CameraClass
+
+  /** \brief Change SpecialEffect
+   *  \param ilevel must be > 0 && <= NUM_SPECIAL_EFFECTS, default 0
+   *  \	1 = no effect
+   *  \ 2 = negative
+   *  \ 3 = black and white
+   *  \ 4 = reddish
+   *  \ 5 = greenish
+   *  \ 6 = blue
+   *  \ 7 = retro
+   */
+  bool
+  changeSpecialEffect(int ilevel);
+
+  /** \brief Change WbMode
+   *  \param ilevel must be > 0 && <= NUM_WB_MODES, default 0
+   *  \ 0 = default
+   *  \ 1 = sunny
+   *  \ 2 = cloudy
+   *  \ 3 = office
+   *  \ 4 = home
+   */
+  bool
+  changeWbMode(int ilevel);
+
+  /** \brief Change AE Levels
+   *  \param ilevel must be -2 to +2, default 0
+   */
+  bool
+  changeAeLevels(int ilevel);
+  
+  /** \brief Enable/Disable  AEC (Auto Exposure Control)
+   *  \param iEnable must be 0(disable) or 1 (enable)
+   */
+  bool
+  changAutoExposurecontrol(int iEnable);
+
+  /** \brief Change AGC Gain
+   *  \param ilevel must be >= 0 and <= 30, default 30
+   */
+  bool
+  changAgcGain(int ilevel);
+
+  /** \brief Change Gainceiling
+   *  \param iGainCeiling must be between GAINCEILING_2X and GAINCEILING_128X
+   */
+  bool
+  changGainceilingSensor(int iGainCeiling);
+
+  /** \brief Enable/Disable Awb GainControl
+   *  \param iEnable must be 0(disable) or 1 (enable)
+   */
+  bool
+  changeAwbGainControl(int iEnable);
+  
+  /** \brief Enable/Disable Gaincontrol
+   *  \param iEnable must be 0(disable) or 1 (enable)
+   */
+  bool
+  changeGaincontrol(int iEnable);
+
+  /** \brief Enable/Disable Testing Colorbar
+   *  \param iEnable must be 0(disable) or 1 (enable)
+   */
+  bool
+  changeColorbar(int iEnable);
+
+  /** \brief Enable/Disable  Whitebalance
+   *  \param iEnable must be 0(disable) or 1 (enable)
+   */
+  bool
+  changeWhitebalance(int iEnable);
+
+  /** \brief Change JPEG Quality 
+   *  \param iEnable iQuality must be within 0 and 63
+   */
+  bool changeQuality(int iQuality);	
+
+  /** \brief Change AEC Value
+   *  \param iEnable iValue must be 0 to 1200, default 0
+   */
+  bool
+  changeAecValue(int iValue);
+
+  /** \brief Enable/Disable Auto/Manual Exposure control
+   *  \param iEnable must be 0(disable) or 1 (enable)
+  */
+  bool
+  changAec2(int iEnable);
+
+  /** \brief Enable/Disable horizontal mirror
+   *  \param iEnable must be 0(disable) or 1 (enable)
+  */
+  bool
+  changHMirror(int iEnable);
+
+  /** \brief Enable/Disable vertical mirror
+   *  \param iEnable must be 0(disable) or 1 (enable)
+  */
+  bool
+  changVFlip(int iEnable);
+
+  /** \brief Enable/Disable black pixel correction
+   *  \param iEnable must be 0(disable) or 1 (enable)
+  */
+  bool
+  changBPC(int iEnable);
+
+  /** \brief Enable/Disable white pixel correction
+   *  \param iEnable must be 0(disable) or 1 (enable)
+  */
+  bool
+  changWPC(int iEnable);
+
+  /** \brief Enable/Disable Denoise Control
+   *  \param iEnable must be 0(disable) or 1 (enable)
+  */
+  bool
+  changDenoise(int iEnable);
+
+  /** \brief Enable/Disable Lenc Correction
```

Comment:
What is Lenc correction?

Reply:
Some questions will be answered here: 
https://github.com/espressif/esp32-camera/blob/e689c3b082985ee7b90198be32d330ce51ac5367/sensors/ov2640.c

And some you need to research in ov2640 sensor doc (Or whatever sensor you have)
https://www.uctronics.com/download/cam_module/OV2640DS.pdf

I remeber the lenc and rawgma had big impact on how low light is being handeled.

Reply:
And some infos can be found here and here:
https://github.com/espressif/esp32-camera/blob/e689c3b082985ee7b90198be32d330ce51ac5367/sensors/private_include/ov2640_settings.h
https://github.com/espressif/esp32-camera/blob/e689c3b082985ee7b90198be32d330ce51ac5367/sensors/private_include/ov2640_regs.h